### DISCUSSION TITLE
how to delete a selection of detailed geometries

### DISCUSSION THREAD

**Discussion by urskaufmann** [QGEP/QGEP#673] [Q&A]
detailed geometries can be deleted with the remove part tool (chap 2.2.6.5 in the QGEP-docu). You have to click on every detail geometry. How can I delete a selection of detailed geometries without clicking every record?

**Comment by ponceta** [comment]
If you mean without removing the corresponding wastewater structures you can't natively on QGIS.

But I'm sure we can add a small action to do this? Or ask for a QGIS core feature?

![Remove_multiple_detailed_geometries](https://user-images.githubusercontent.com/8614223/119449043-670f4300-bd32-11eb-8bee-5ad614b16067.gif)

**Comment by urskaufmann** [comment]
@ponceta : Yes, without deleting the wastewater structure. Otherwise it's easy.  What you show in the video is the thing described in chap. 2.2.6.5. It works also without selecting the polygon before.
My problem: I got Geonis-Data - and there you have a detailgeometrie with every reach and every manhole...

**Comment by sjib** [comment]
@urskaufmann Can't you just delete the detailed_geometry in pg_admin in the database. 
Something like:
`DELETE detailed_geometry_geometry FROM qgep_od.reach WHERE ..`. (and make a selection, if you do not want to DELETE all)?

**Comment by urskaufmann** [comment]
Something like (but do not know the right expressions):
UPDATE qgep_od.wastewater_structure
SET detail_geometry_geometry = '' 
WHERE fk_main_wastewater_node IS NULL

**Comment by sjib** [comment]
`UPDATE qgep_od.wastewater_structure SET detail_geometry_geometry = NULL WHERE fk_main_wastewater_node IS NULL` funktioniert bei mir.

**Comment by ponceta** [comment]
@urskaufmann Does this solve your problem? Or do we need some graphical tool?

**Comment by urskaufmann** [comment]
Yes, if there is a simply WHERE definition and if I am a database admin.
No, if I am a qgep-user. 
It should be possible to delete all detail_geometries of a selection in the layer od_wastewater_structure.
And to make a selection, there should be more attributes (also from related tables as manhole-function, specialstructure-function) in the attribute table, which means, it should be a view.
